tests/preprocessing/test_data.py

```diff
@@ -646,7 +632,12 @@ def test_transformer_params(self):
         assert pf._transformer.interaction_only is pf.interaction_only
         assert pf._transformer.include_bias is pf.include_bias
 
-    @pytest.mark.parametrize("daskify", [True, False])
+    mark = pytest.mark.xfail(
+        DASK_EXPR_ENABLED,
+        reason="dask-expr: NotImplementedError in assert_eq_df(res_df.iloc[:, 1:], frame, check_dtype=False)",
+    )
+
+    @pytest.mark.parametrize("daskify", [pytest.param(True, marks=mark), False])
     def test_df_transform_index(self, daskify):
```

Comment:
@phofl 

<details>
<summary>traceback</summary>

```
___________________________ TestPolynomialFeatures.test_df_transform_index[True] ___________________________

self = <tests.preprocessing.test_data.TestPolynomialFeatures object at 0x7f0c5bf00bb0>, daskify = True

    @pytest.mark.parametrize("daskify", [pytest.param(True, marks=mark), False])
    def test_df_transform_index(self, daskify):
        frame = copy(df)
        if not daskify:
            frame = frame.compute()
        frame = frame.sample(frac=1.0)
        res_df = dpp.PolynomialFeatures(
            preserve_dataframe=True, degree=1
        ).fit_transform(frame)
>       assert_eq_df(res_df.iloc[:, 1:], frame, check_dtype=False)

tests/preprocessing/test_data.py:649:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <dask_expr._indexing.ILocIndexer object at 0x7f0c691fb400>
key = (slice(None, None, None), slice(1, None, None))

    def __getitem__(self, key):
        msg = (
            "'DataFrame.iloc' only supports selecting columns. "
            "It must be used like 'df.iloc[:, column_indexer]'."
        )
        if not isinstance(key, tuple):
            raise NotImplementedError(msg)

        if len(key) > 2:
            raise ValueError("Too many indexers")

        iindexer, cindexer = key

        if iindexer != slice(None):
            raise NotImplementedError(msg)

        if len(self.obj.columns) == len(set(self.obj.columns)):
            col_names = self.obj.columns[cindexer]
            if not is_scalar(col_names):
                col_names = list(col_names)
            return new_collection(Projection(self.obj, col_names))
        else:
>           raise NotImplementedError
E           NotImplementedError

../../mambaforge/envs/dask-ml/lib/python3.10/site-packages/dask_expr/_indexing.py:51: NotImplementedError
```

</details>

Reply:
does the DataFRame have duplicated columns? Then yes, we don't care for now

Reply:
Ah, okay the output `res_df` has duplicate columns, but seems expected as that's what the iloc is for, but you're saying b/c of that we don't care right now because it's not implemented. 

```python
(Pdb) p frame.columns
Index(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',
       '13', '14', '15', '16', '17', '18', '19'],
      dtype='object')
(Pdb) p res_df.columns
Index(['1', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',
       '13', '14', '15', '16', '17', '18', '19'],
      dtype='object')
```